datadog/dogstatsd/base.py

```diff
@@ -289,9 +289,12 @@ def _send_to_server(self, packet):
         except socket.timeout:
             # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
             return
-        except socket.error:
+        except (socket.error, socket.herror, socket.gaierror) as se:
             log.info("Error submitting packet, dropping the packet and closing the socket")
             self.close_socket()
```

Comment:
Should we be closing the socket in the second `except` clause as well?

Reply:
Not sure, `# If set, use socket directly`, would the last exception cover the case of not using the socket and thus not needing to close it? Or am I on the wrong track there?